arithmetization/src/main/java/net/consensys/linea/zktracer/module/ec_data/EcDataOperation.java

```diff
@@ -0,0 +1,348 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.zktracer.module.ec_data;
+
+import static net.consensys.linea.zktracer.types.Containers.repeat;
+import static net.consensys.linea.zktracer.types.Conversions.leftPadTo;
+
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.base.Preconditions;
+import lombok.Getter;
+import lombok.experimental.Accessors;
+import net.consensys.linea.zktracer.module.ext.Ext;
+import net.consensys.linea.zktracer.module.wcp.Wcp;
+import net.consensys.linea.zktracer.opcode.OpCode;
+import net.consensys.linea.zktracer.types.EWord;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+
+@Accessors(fluent = true)
+public class EcDataOperation {
+  private static final int EC_RECOVER = 1;
+  private static final int EC_ADD = 6;
+  private static final int EC_MUL = 7;
+  private static final int EC_PAIRING = 8;
+  private static final Set<Integer> EC_TYPES = Set.of(EC_RECOVER, EC_ADD, EC_MUL, EC_PAIRING);
+
+  private static final EWord P =
+      EWord.ofHexString("0x30644e72e131a029b85045b68181585d97816a916871ca8d3c208c16d87cfd47");
+  private static final EWord Q =
+      EWord.ofHexString("0x30644E72E131A029B85045B68181585D2833E84879B9709143E1F593F0000001");
+  private static final EWord SECP256_K1N =
+      EWord.ofHexString("0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f");
+
+  private final Wcp wcp;
+  private final Ext ext;
+
+  private final int contextNumber;
+  private final int contextNumberDelta;
+  private final Bytes input;
+
+  private final int ecType;
+  @Getter private final int nRows;
+  /** -1 if no switch off (i.e. preliminary checks passed) */
+  private int hurdleSwitchOffRow;
+
+  private final List<Boolean> comparisons;
+  private final List<Boolean> equalities;
+  private final List<Bytes> squares;
+  private final List<Bytes> cubes;
+  private final List<Bytes> limbs;
+
+  // WCP interaction
+  private final List<Bytes> wcpArg1;
+  private final List<Bytes> wcpArg2;
+  private final List<OpCode> wcpInst;
+  private final List<Boolean> wcpRes;
+
+  // EXT interaction
+  private final List<Bytes> extArg1;
+  private final List<Bytes> extArg2;
+  private final List<Bytes> extArg3;
+  private final List<OpCode> extInst;
+  private final List<Bytes> extRes;
+
+  // pairing-specific
+  private int thisIsNotOnG2Index;
+  private final int nPairings;
+
+  private static int addressToEcType(final Address target) {
```

Comment:
use switch ?

Reply:
Not sure of how the switch behaves w.r.t. object comparison (== or equals?)